### PR TITLE
Ensure that default value is present in enum store.

### PR DESCRIPTION
**Pull request by toregge**
@geirst : please review
@baldersheim : FYI

### CODE REVIEW

File: searchlib/src/vespa/searchlib/attribute/enumstore.hpp [hunk 1]
```diff
@@ -78,12 +79,18 @@ EnumStoreT<EntryT>::EnumStoreT(bool has_postings, const DictionaryConfig& dict_c
       _is_folded(dict_cfg.getMatch() == DictionaryConfig::Match::UNCASED),
       _comparator(_store.get_data_store()),
       _foldedComparator(make_optionally_folded_comparator(is_folded())),
-      _compaction_spec()
+      _compaction_spec(),
+      _default_value(attribute::getUndefined<EntryT>()),
+      _default_value_ref()
 {
+    if constexpr (std::is_same_v<const char *, EntryT>) {
```
Comment (baldersheim) [inline searchlib/src/vespa/searchlib/attribute/enumstore.hpp:86]: Add `attribute::getUndefined<EntryT>` for const char too ?

File: searchlib/src/tests/attribute/enumstore/enumstore_test.cpp [hunk 2]
```diff
@@ -188,7 +188,7 @@ TEST(EnumStoreTest, test_find_folded_on_string_enum_store)
     for (std::string &str : unique) {
         EnumIndex idx = ses.insert(str.c_str());
         indices.push_back(idx);
-        EXPECT_EQ(1u, ses.get_ref_count(idx));
+        EXPECT_EQ((str == "") ? 2u : 1u, ses.get_ref_count(idx));
```
Comment (baldersheim) [inline searchlib/src/tests/attribute/enumstore/enumstore_test.cpp:191]: Use `attribute::getUndefined<EntryT>` or `attribute::hasUndefined` ?

**Review by baldersheim — COMMENTED** [review]
Not sure if I understand the rational for all of this, but that we can discuss over the weekend.

**Review by geirst — DISMISSED** [review]
👍

**Review by geirst — DISMISSED** [review]
👍

**Pull request merged**